R/snapshot.R

```diff
@@ -969,20 +970,17 @@ renv_snapshot_filter_report_missing <- function(missing, type) {
     # do nothing
   } else if (choice == "install") {
     install(missing, prompt = FALSE)
-    # User will only see this in exceptional circumstances as it is
-    # caught once by renv_lockfile_create()
-    stop(errorCondition(
-      message = "Failed to restart snapshotting after install",
-      class = "renv_recompute_records"
-    ))
+    renv_condition_signal("renv_recompute_records")
```

Comment:
FWIW it feels like we should be setting a flag or something indicating whether we would prompt the user here.

Reply:
I think this needs to be a stop, otherwise if `install()` doesn't get everything the first time, and the user tries to `install()` again, then it will appear to succeed.  I don't think the previous behaviour was that great either, but at least you got a clear error.

Reply:
I think I have the right solution now:

- Move the lockfile creation impl into a separate function
- Invoke that from the top level in a `withRestarts` handler
- Check for relevant restarts when prompting the user, and only prompt to install if such a restart exists

The action at a distance is sort of hard to reason about, but I believe it's correct.